Test/Utils/ContrailAPI/ConfigureDNS.Tests.ps1

```diff
@@ -0,0 +1,132 @@
+
+. $PSScriptRoot\ConfigureDNS.ps1
+. $PSScriptRoot\..\ContrailUtils.ps1
+. $PSScriptRoot\..\..\..\CIScripts\Testenv\Testenv.ps1
+
+
+# TODO: Those tests run on working Controller.
+#       Most probably they need to be rewrote
+#       to use some fake.
+# NOTE: Because of the above they should not be run automatically
+Describe 'Configure DNS API' {
+    BeforeAll {
+        $OpenStackConfig = Read-OpenStackConfig -Path $TestenvConfFile
+        $ControllerConfig = Read-ControllerConfig -Path $TestenvConfFile
+
+        [Diagnostics.CodeAnalysis.SuppressMessageAttribute(
+            "PSUseDeclaredVarsMoreThanAssignments",
+            "ContrailUrl",
+            Justification="It's actually used."
+        )]
+        $ContrailUrl = $ControllerConfig.RestApiUrl()
+
+        [Diagnostics.CodeAnalysis.SuppressMessageAttribute(
+            "PSUseDeclaredVarsMoreThanAssignments",
+            "AuthToken",
+            Justification="It's actually used."
+        )]
+        $AuthToken = Get-AccessTokenFromKeystone `
+            -AuthUrl $OpenStackConfig.AuthUrl() `
+            -Username $OpenStackConfig.Username `
+            -Password $OpenStackConfig.Password `
+            -Tenant $OpenStackConfig.Project
+    }
+
+    It 'adds and removes "empty" contrail DNS server' {
+        $ServerUUID = Add-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script"
+        Remove-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSServerUuid $ServerUUID
+    }
+
+    It 'adds and removes DNS server records' {
+        $ServerUUID = Add-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script"
+
+        $Record1Data = [VirtualDNSRecordData]::new("host1", "1.2.3.4", "A")
+        $Record1UUID = Add-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script" -VirtualDNSRecordData $Record1Data
+
+        $Record2Data = [VirtualDNSRecordData]::new("host2", "1.2.3.5", "A")
+        $Record2UUID = Add-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script" -VirtualDNSRecordData $Record2Data
+
+        Remove-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSRecordUuid $Record1UUID
+        Remove-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSRecordUuid $Record2UUID
+
+        Remove-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSServerUuid $ServerUUID
+    }
+
+    It 'breaks when wrong dns mode specified' {
+        { Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'wrongdnsmode' } | Should -Throw
+    }
+
+    It 'sets DNS mode to none' {
+        Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
```

Comment:
Maybe we should create `Context` blocks, like:

* Context "DNS server creation and removal"
* Context "Setting DNS modes"

Reply:
We can further split "Setting DNS modes" by DNS mode types maybe?